### PR TITLE
Swap svn for GitHub API calls

### PR DESCRIPTION
**Pull request by harveysanders**
- WIP - Swap svn for GitHub API

TODO:
- [ ] Ensure tests work with real creds
- [ ] Mock GitHub API to fix tests

### CODE REVIEW

File: controller/github-api.js [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+
+/** @typedef {import("@octokit/openapi-types/types").components  } GitHubComponents */
+/** @typedef {GitHubComponents["schemas"]["content-directory"] | GitHubComponents["schemas"]["content-file"] } GetContentResp */
+
+class GithubAPI {
+  /**
+   *
+   * @param {string} token
+   * @returns
+   */
```
Comment (ptbarnum4) [inline controller/github-api.js:14]:
```suggestion
  /** @param {string} token */
```

File: controller/github-api.js [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+      auth: `token ${token}`,
+      userAgent: `opspark_tool@${version}`
+    });
+  }
+
+  /**
+   *
+   * @param {string} url
+   * @returns {{owner: string, repo: string}}
+   */
```
Comment (ptbarnum4) [inline controller/github-api.js:26]:
```suggestion
  /**
   * Parse the owner/organization and repo name from URL
   * @param {string} url
   * @returns {{owner: string, repo: string}}
   * 
   * @example
   * parseRepoUrl('https://github.com/OperationSpark/opspark');
   * // { owner: 'OperationSpark', repo: 'opspark' }
   */
```

File: controller/github-api.js [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+    return {
+      owner: parsed.pathname.split('/')[1],
+      repo: parsed.pathname.split('/')[2]
+    };
+  }
+  /**
+   *
+   * @param {string} url
+   * @param {string} directory
+   */
```
Comment (ptbarnum4) [inline controller/github-api.js:38]:
```suggestion
  /**
   * Download tests for a project into specified directory
   * @param {string} url
   * @param {string} directory
   */
```

File: controller/test.js [hunk 4]
```diff
@@ -1,45 +1,30 @@
...
-//   githubDir = fs
-//     .readdirSync(rootDirectory)
-//     .filter(dir => /[\w]+\.github\.io/.test(dir))[0];
-//   rootDirectory = `${rootDirectory}/${githubDir}`;
-// }
 const projectsDirectory = `${rootDirectory}/projects`;
 
 // Start of test command
 // Runs the listProjectsOf function from projects to select project
 // that user wants to be tested
```
Comment (ptbarnum4) [inline controller/test.js:24]:
```suggestion
/** 
  * Start of test command runs the listProjectsOf function from projects 
  * to select project that user wants to be tested
  */
```

File: install.sh
Comment (ptbarnum4) [inline install.sh:1]: is this file still needed? 

Having this script prevents cross-platform compatibility. I think we can specify the node version in the package.json and just have a postinstall for the global mocha install 

```json
{
  ...
  "scripts": {
    ...
    "postinstall": "npm I -g mocha"
    ...
  }
  ...
}
```

File: test/test-greenlight.js [hunk 5]
```diff
@@ -1,16 +1,8 @@
-/* global describe it expect before beforeEach afterEach after */
+/* global describe it before beforeEach afterEach after */
```
Comment (ptbarnum4) [inline test/test-greenlight.js:1]: I think this can be solved by installing the types

`npm i -D @types/mocha @type/chai`

**Review by ptbarnum4 — APPROVED** [review]
looks great!

**Review by alex-aaron — APPROVED** [review]
This all makes sense!

**Comment by ptbarnum4** [comment]
@harveysanders [is this the flaky test](https://github.com/OperationSpark/opspark/actions/runs/7791818343/job/21248656126#step:5:575)

**Comment by harveysanders** [comment]
Yeah

**Comment by harveysanders** [comment]
> @harveysanders [is this the flaky test](https://github.com/OperationSpark/opspark/actions/runs/7791818343/job/21248656126#step:5:575)

@ptbarnum4 yeah, they should be fixed on [this commit](https://github.com/OperationSpark/opspark/pull/78/commits/37b44592de043a996bdaeb61d2f790ab5d6fff76) now
